src/de/jost_net/JVerein/server/DDLTool/Updates/Update0422.java

```diff
@@ -0,0 +1,124 @@
+ *  the GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License along with this program.  If not, 
+ * see <http://www.gnu.org/licenses/>.
+ * 
+ **********************************************************************/
+package de.jost_net.JVerein.server.DDLTool.Updates;
+
+import java.sql.Connection;
+import java.sql.ResultSet;
+import java.sql.Statement;
+import java.util.ArrayList;
+import java.util.HashMap;
+
+import de.jost_net.JVerein.gui.dialogs.AuswahlDialog;
+import de.jost_net.JVerein.server.DDLTool.AbstractDDLUpdate;
+import de.jost_net.JVerein.server.DDLTool.Column;
+
+import de.willuhn.jameica.gui.dialogs.ListDialog;
+import de.willuhn.jameica.system.OperationCanceledException;
+import de.willuhn.logging.Logger;
+import de.willuhn.util.ApplicationException;
+import de.willuhn.util.ProgressMonitor;
+
+public class Update0422 extends AbstractDDLUpdate {
+  public Update0422(String driver, ProgressMonitor monitor, Connection conn) {
+    super(driver, monitor, conn);
+  }
+
+  @Override
+  public void run() throws ApplicationException {
+    HashMap<String, String> auswahl_moeglichkeiten = new HashMap<String, String>();
+    auswahl_moeglichkeiten.put("Buchungsnummer", "id");
+    auswahl_moeglichkeiten.put("Name", "name");
+    auswahl_moeglichkeiten.put("Verwendungszweck", "zweck");
+    auswahl_moeglichkeiten.put("Auszugsnummer", "auszugsnummer");
+    auswahl_moeglichkeiten.put("Blattnummer", "blattnummer");
+
+    AuswahlDialog d = new AuswahlDialog(ListDialog.POSITION_CENTER,
+        new ArrayList<String>(auswahl_moeglichkeiten.keySet()), "Belegnummern Import", "");
+    d.setText(
+        "In der neueste jVerein Version besteht die M�glichkeit manuelle Belegnummern anstatt "
+            + "der Buchungsnummer aus der Datenbank zu verwenden.\n\nFalls du bisher schon Belegnummern verwendet hast, "
+            + "kannst du im folgenden Dialog ausw�hlen, aus welchem bisherigen Buchungsfeld diese importiert werden sollen. "
+            + "Falls du dieses Feature nicht nutzen m�chtest, w�hle \"Buchungsnummer\" aus.\n\n");
+    d.setSize(400, 300);
+
+    try {
+      String choice = (String) d.open();
+      String feld = auswahl_moeglichkeiten.get(choice);
+
+      // neue Felder hinzuf�gen und dabei VersionsNr erst einmal nicht ver�ndern!
+      execute(addColumn("einstellung",
+          new Column("verwendebelegnummer", COLTYPE.BOOLEAN, 0, "FALSE", false, false)), false);
+      execute(
+          addColumn("einstellung",
+              new Column("belegnummer_pro_konto", COLTYPE.BOOLEAN, 0, "TRUE", false, false)),
+          false);
+      execute(addColumn("einstellung",
+          new Column("belegnummer_pro_jahr", COLTYPE.BOOLEAN, 0, "TRUE", false, false)), false);
+      execute(
+          addColumn("buchung", new Column("belegnummer", COLTYPE.INTEGER, 19, "-1", false, false)),
```

Comment:
Warum -1, wenn neue Buchungen bei deaktiviertem Feature 0 bekommen?

Reply:
Steht in Zeile 100. Um zu prüfen, dass die hinterlegten Belegnummern nicht konvertiert werden können.